### PR TITLE
Merge v2.5.2 main into pre-staging

### PR DESCRIPTION
**Pull request by dbobrenko**

### CODE REVIEW

File: prompting/base/prompting_miner.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 from prompting.protocol import StreamPromptingSynapse
 from prompting.base.miner import BaseStreamMinerNeuron
 from datetime import datetime
-
+from typing import List, Dict
```
Comment (Hollyqui) [inline prompting/base/prompting_miner.py:26]: python now has built-in types which are generally preferred over typing

File: prompting/miners/langchain_miner.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
+# The MIT License (MIT)
```
Comment (Hollyqui) [inline prompting/miners/langchain_miner.py:1]: Why we're adding a deprecated miner to the codebase (that wasn't there before)? I'm sure there's a good reason, just not clear to me

File: prompting/base/prompting_miner.py (hunk 1 above)
Comment (dbobrenko) [reply prompting/base/prompting_miner.py:26]: Yeah, this is some old code from python 3.8.
It's already on main tho, this PR is just to keep pre-staging up to date.

File: prompting/miners/langchain_miner.py (hunk 2 above)
Comment (dbobrenko) [reply prompting/miners/langchain_miner.py:1]: This is also some old code on main, and probably this miner should be deleted

**Review by Hollyqui — APPROVED** [review]
Check comments, otherwise LGTM

**Pull request merged**